packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -0,0 +1,317 @@
+    let flag = false;
+    const key = event.key;
+
+    const printableCharacter = () => {
+      if (key === '*') {
+        expandAllSiblings(nodeId);
+        flag = true;
+      } else if (isPrintableCharacter(key)) {
+        setFocusByFirstCharacter(nodeId, key);
+        flag = true;
+      }
+    };
+
+    if (event.altKey || event.ctrlKey || event.metaKey) {
+      return;
+    }
+    if (event.shift) {
+      if (key === ' ' || key === 'Enter') {
+        event.stopPropagation();
+      } else if (isPrintableCharacter(key)) {
+        printableCharacter();
+      }
+    } else {
+      switch (key) {
+        case 'Enter':
+        case ' ':
+          if (nodeRef.current === event.currentTarget && expandable) {
+            toggle();
+            flag = true;
+          }
+          event.stopPropagation();
+          break;
+        case 'ArrowDown':
+          focusNextNode(nodeId);
+          flag = true;
+          break;
+        case 'ArrowUp':
+          focusPreviousNode(nodeId);
+          flag = true;
+          break;
+        case 'ArrowRight':
+          if (expandable) {
+            if (expanded) {
+              focusNextNode(nodeId);
+            } else {
+              toggle();
+            }
+          }
+          flag = true;
+          break;
+        case 'ArrowLeft':
+          handleLeftArrow(nodeId, event);
+          break;
+        case 'Home':
+          focusFirstNode();
+          flag = true;
+          break;
+        case 'End':
+          focusLastNode();
+          flag = true;
+          break;
+        default:
+          if (isPrintableCharacter(key)) {
+            printableCharacter();
+          }
+      }
+    }
+
+    if (flag) {
+      event.preventDefault();
+      event.stopPropagation();
+    }
+
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+  };
+
+  const handleFocus = event => {
+    if (!focused && tabable) {
+      focus(nodeId);
+    }
+
+    if (onFocus) {
+      onFocus(event);
+    }
+  };
+
+  React.useEffect(() => {
+    const childIds = React.Children.map(children, child => child.props.nodeId);
+    if (handleNodeMap) {
+      handleNodeMap(nodeId, childIds);
+    }
+  }, [children, nodeId, handleNodeMap]);
+
+  React.useEffect(() => {
+    if (handleFirstChars && label) {
+      handleFirstChars(nodeId, label.substring(0, 1).toLowerCase());
+    }
+  }, [handleFirstChars, nodeId, label]);
+
+  React.useEffect(() => {
+    if (focused) {
+      nodeRef.current.focus();
+    }
+  }, [focused]);
+
+  return (
+    <li
+      className={clsx(classes.root, className)}
+      role="treeitem"
+      onKeyDown={handleKeyDown}
+      onFocus={handleFocus}
+      aria-expanded={expandable ? expanded : null}
+      ref={handleRef}
+      tabIndex={tabable ? 0 : -1}
+      {...other}
+    >
+      <div className={classes.content} onClick={handleClick} ref={contentRef}>
+        {stateIconsProvided ? <div className={classes.iconContainer}>{stateIcon}</div> : null}
```

Comment:
```suggestion
        {stateIconsProvided && <div className={classes.iconContainer}>{stateIcon}</div>}
```
Is there a reason this syntax isn't used?
https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
Not really, was just following what I found in the code. Looks like the only time the above syntax is used is in CardHeader vs SvgIcon, LinearProgress, GridListTileBar, SnackbarContent. I'm happy with either syntax. Normally I prefer the one you suggested. Do either of you mind @oliviertassinari @eps1lon

Reply:
Second one is idiomatic react for conditional rendering. If this is the intention then we should use it. If the ternary is used for a reason other than conditional rendering it should be documented.

Reply:
React documents both: https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator. I don't have a preference. I used to use `&&` when first using React, nowadays, I only use ternaries. Maybe we should use `&&` because it yields smaller code? However, if we use `&&` we need to make sure we compare with boolean! e.g. 0 && <span /> will display 0 🙃

Reply:
It's not that important in the end IMO. Personally I would stick with strict equality checks and ternaries but as long as one pattern doesn't introduce more bugs than the other it really doesn't matter. 

Or we introduce something like Bikeshed-Sundays were we batch those discussions :smile:

Reply:
Valid point, it doesn't change anthing at the end of the day. As long as we use the same pattern everywhere, that we only have this discussion once, I'm happy. So all in in ternaries?

Reply:
> Or we introduce something like Bikeshed-Sundays were we batch those discussions

@eps1lon 😆 I’m not sure I like the name though. What about `Bikeshed Sundays` it’s soooo much cleaner 😉